packages/trip-form/src/MetroModeSelector/index.tsx

```diff
@@ -0,0 +1,346 @@
+        blockPointerEvents: false,
+        restMs: 500,
+        buffer: 0
+      })
+    }),
+    useClick(context),
+    useRole(context),
+    useDismiss(context)
+  ]);
+
+  const renderDropdown = open && modeButton.enabled;
+  const interactionProps = getReferenceProps();
+
+  // ARIA roles are added by the `useRole` hook.
+  // Remove the aria-controls, aria-expanded, and aria-haspopup props from the label, they will
+  // instead be passed to the button for keyboard/screen reader users to trigger the popup.
+  const {
+    "aria-controls": ariaControls,
+    "aria-expanded": ariaExpanded,
+    "aria-haspopup": ariaHasPopup,
+    ...labelInteractionProps
+  } = interactionProps;
+
+  const checkboxId = `metro-mode-selector-mode-${id}`;
+
+  return (
+    <ModeButtonWrapper>
+      {/* Basic checkbox that states whether a mode is selected. */}
+      <input
+        checked={modeButton.enabled ?? undefined}
+        id={checkboxId}
+        onChange={onToggle}
+        type="checkbox"
+        aria-label={modeButton.label}
+      />
+      {/* Label for the above checkbox, placed right after, so that CSS applies based on checkbox state. */}
+      {/* eslint-disable-next-line jsx-a11y/label-has-associated-control */}
+      <label
+        // This library relies on prop spreading
+        // eslint-disable-next-line react/jsx-props-no-spreading
+        {...labelInteractionProps}
+        htmlFor={checkboxId}
+        // This will trigger mouse effects such as showing popup on hover of on check.
+        ref={reference}
+        title={modeButton.label}
+      >
+        <modeButton.Icon role="none" size={32} />
+        <InvisibleA11yLabel>{modeButton.label}</InvisibleA11yLabel>
+      </label>
+      <button
+        // eslint-disable-next-line react/jsx-props-no-spreading
+        {...interactionProps}
+        // Disable button if mode is not checked (but keep in DOM for screen reader awareness)
+        disabled={!modeButton.enabled}
+        // Required by linter settings
+        type="button"
+      >
+        <span role="none">
+          {open ? <CaretUp size={14} /> : <CaretDown size={14} />}
+        </span>
+        <InvisibleA11yLabel>
+          <FormattedMessage
+            defaultMessage={defaultMessages["otpUi.ModeSelector.settingsLabel"]}
+            description="Label for the button to open settings for a travel mode."
+            id="otpUi.ModeSelector.settingsLabel"
+            values={{ mode: modeButton.label }}
+          />
+        </InvisibleA11yLabel>
+      </button>
+      {renderDropdown && (
+        <FloatingFocusManager context={context}>
+          <HoverPanel
+            // This library relies on prop spreading
+            // eslint-disable-next-line react/jsx-props-no-spreading
+            {...getFloatingProps()}
+            // Matches ID on Header element in SubSettingsPane
+            aria-labelledby={`metro-mode-selector-${modeButton.key}-button-label`}
+            ref={floating}
```

Comment:
This one should be fixed: When mouse-hovering out of a label that has the popup open, the focus is set on that label, even if keyboard focus was on a different element before.

Reply:
Could you revert https://github.com/opentripplanner/otp-ui/pull/555/commits/ab66a4db693cf65c0ab3dfd96e8946475852204f? I think that change has more side effects than the initial behavior.

Reply:
I'm leaning towards disabling hovering for the duration that a keyboard-triggered popup is open. I think we could do that in a separate PR but let me run that with @demory.

Reply:
@daniel-heppner-ibigroup See https://github.com/ibi-group/otp-ui/pull/16 that addresses the issues above.